.changelog/5070.cfg.md

```diff
@@ -0,0 +1,105 @@
+Configuration changes
+
+The node now supports a configuration file that holds all the settings
+that were previously available via command-line options.
+The command-line options themselves have been removed in favor of
+the config file.
+
+The config file supports environment variable substitution using
+the `envsubst` package
+([usage information](https://github.com/a8m/envsubst#docs)).
+
+The following options have been removed:
+
+- `consensus.tendermint.mode` in favor of using the global node mode (`mode`).
+
+- `runtime.mode` in favor of using the global node mode (`mode`).
+
+- `worker.client.port` as it is no longer needed.
+
+- `worker.registration.force_register` as it is deprecated.
+
+The following options have been renamed:
```

Comment:
Given that there have been a lot of changes, can we also make a migration subcommand? So that we can include that in the upgrade instructions.

Reply:
As agreed on Slack, we're going to make this in a separate PR later.